src/TestFramework/PhpUnit/Coverage/XmlCoverageParser.php

```diff
@@ -49,31 +49,20 @@
  */
 final class XmlCoverageParser
 {
-    private $provider;
-
-    public function __construct(SourceFileInfoProvider $provider)
+    public function __construct()
     {
-        $this->provider = $provider;
     }
 
-    public function parse(): SourceFileData
+    public function parse(SourceFileInfoProvider $provider): SourceFileData
     {
         return new SourceFileData(
-            $this->provider->provideFileInfo(),
-            $this->lazilyRetrieveCoverageReport(
-                $this->provider->provideXPath()
-            )
+            $provider->provideFileInfo(),
+            (static function (SafeDOMXPath $xPath) {
+                yield self::retrieveCoverageReport($xPath);
+            })($provider->provideXPath())
```

Comment:
IIRC one-off closures are slightly slower than plain old static functions, but this might be not hot enough place to show it. Not mentioning that adding return type declaration to closures is an annoyance.

Reply:
It was mainly to get it of `lazilyRetrieveCoverageReport()` tbh... I was not fond of the name but could not come up with a better one and since it's a one-liner method...

Reply:
Well, for a reason, see? 

There's another. We don't use Psalm yet, but if we'd ever do, this will be another thing to document with types. Where `lazilyRetrieveCoverageReport()` was already documented properly.